### PR TITLE
Bug 1992004: Fix flaky e2e test by updating a workaround for rate limit error by GitHub

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://bugzilla.redhat.com/show_bug.cgi?id=1992004

**Analysis / Root cause**: 
There was an e2e code which was only called when the import from git flow receives a rate limited exceeded error from GitHub.

In this case the test automatically selects the right builder image. The `[aria-label="Perl"]` selector to select the right builder image doesn't work anymore since (my) PR #9360

**Solution Description**: 
Fix the select builder image method and use a new `[data-test="card perl"]` selector.

**Screen shots / Gifs for design review**: 
Before:
before

After:
after

**Unit test coverage report**: 
Not touched

**Test setup:**
Open cypress browser with `yarn test-cypress-dev-console` and run the `create-from-git.feature`.

To ensure that the code is called you can remove the if condition around `selectBuilderImageForGitUrl` in `git-page.ts`

See https://github.com/openshift/console/pull/9793#pullrequestreview-728371967

**Browser conformance**: 
Not relevant

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/support/pages/add-flow/git-page.ts [hunk 1]
```diff
@@ -235,15 +235,7 @@ export const gitPage = {
...
       .and('not.include.text', messages.addFlow.buildDeployMessage);
-    cy.get('body').then(($body) => {
-      if (
-        $body.text().includes(messages.addFlow.rateLimitExceeded) ||
-        $body.find('[aria-label="Warning Alert"]').length ||
-        $body.text().includes(messages.addFlow.privateGitRepoMessage)
-      ) {
-        gitPage.selectBuilderImageForGitUrl(gitUrl);
-      }
-    });
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/support/pages/add-flow/git-page.ts:246]: This code is needed, because if we get rate limitation, then only this line ("gitPage.selectBuilderImageForGitUrl(gitUrl);") should execute

Comment (jerolimov) [reply frontend/packages/dev-console/integration-tests/support/pages/add-flow/git-page.ts:246]: My idea is that we call `selectBuilderImageForGitUrl` always so that we notice if the test breaks again in the future. But when you don't like the idea I can restore that we call this only in this edge case which mostly happen on the CI server (only)

Comment (jerolimov) [reply frontend/packages/dev-console/integration-tests/support/pages/add-flow/git-page.ts]: Reverted this as it is not required for our fix. Thanks for the quick feedback.

**Review by dtaylor113 — APPROVED** [review]
/lgtm
